src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1205,6 +1206,21 @@ public <E> void assertSatisfiesExactlyInAnyOrder(AssertionInfo info, Iterable<?
       throw failures.failure(info, shouldSatisfyExactlyInAnyOrder(actual));
   }
 
+  public <E> void assertSatisfiesOnlyOnce(AssertionInfo info, Iterable<? extends E> actual,
+                                          Consumer<? super E> requirements) {
+    assertNotNull(info, actual);
+    requireNonNull(requirements, "The Consumer<? super E> expressing the requirements must not be null");
+
+    long countOfSatisfactions = stream(actual)
+                                              .filter(byPassingAssertions(requirements))
+                                              .count();
```

Comment:
As per my comments on the error message, we want to capture the elements satisfying the requirements to report them if more than one did